utils/splunk/splunk_helper.py

```diff
@@ -38,6 +40,40 @@ def auth_session(self):
         session_key = response_json["sessionKey"]
         self.requests_session.headers.update({'Authentication': "Splunk %s" % session_key})
 
+    def create_auth_token(self, token):
+        self.log.debug("Creating a new authentication token")
+        token_path = '/services/authorization/tokens?output_mode=json'
+        name = self.instance_config.username
+        audience = self.instance_config.audience
+        expiry_days = self.instance_config.token_expiration_days
+        payload = {'name': name, 'audience': audience, 'expires_on': "+{}d".format(str(expiry_days))}
+        self.requests_session.headers.update({'Authorization': "Bearer %s" % token})
+        response = self._do_post(token_path, payload, self.instance_config.default_request_timeout_seconds)
+        response.raise_for_status()
+        response_json = response.json()
+
+        new_token = response_json.get("entry")[0].get("content").get("token")
+        self.requests_session.headers.update({'Authorization': "Bearer %s" % new_token})
+        return new_token
+
+    def is_token_valid(self, token):
+        current_time = self._current_time()
```

Comment:
Are these timestamps all compared in UTC / same timezone? Trying to avoid timestamp timezone issues.

Reply:
changed.